bpf/bpf_overlay.c

```diff
@@ -673,6 +674,21 @@ int cil_from_overlay(struct __ctx_buff *ctx)
 		goto out;
 	}
 
+#if defined(ENABLE_WIREGUARD)
+	/* When wireguard is enabled we should drop any traffic coming through the tunnel
+	 * that previously wasn't marked as decrypted by cilium.
+	 */
+	if (CONFIG(encryption_strict_ingress) && ctx->mark != MARK_MAGIC_DECRYPT) {
+		ret = DROP_UNENCRYPTED_TRAFFIC;
+		goto out;
+	}
+	/* We only needed the mark to decide if we need to drop the packet here.
+	 * To not cause any further collision with the `decrypted` variable,
+	 * clear the mark.
+	 */
+	ctx->mark = 0;
```

Comment:
Q: Should we completely clearing the mark here instead of removing the decrypt bit?

Reply:
I was under the impression that there shouldn't be anything before that we could override, but might be wrong on that.